libraries/irule_diff.rb

```diff
@@ -0,0 +1,51 @@
+require 'set'
+
+# Diff 2 hashes to show the added, changed, and removed keys
+class IRuleDiff
+  class << self
+    MAX_IRULES = 100
+
+    def diff_hashes(target, current)
+      target_keys = Set.new target.keys
+      current_keys = Set.new current.keys
+      added = target_keys - current_keys
+      removed = current_keys - target_keys
+      changed = target.keys.select { |k| !current[k].nil? && target[k] != current[k] }
+      x = [added, changed, removed, target]
+      x
```

Comment:
These last two lines could be cleaned up to just `[added, changed, removed, target]` I think